### PR TITLE
clientv3/integration: add 'TestUserErrorAuth'

### PR DESCRIPTION
**Pull request by gyuho**
Since we don't have any test cases around 'wrong' id or password

### CODE REVIEW

File: clientv3/integration/user_test.go [hunk 1]
```diff
@@ -51,3 +52,51 @@ func TestUserError(t *testing.T) {
...
+	authapi := clus.RandClient()
+	authSetupRoot(t, authapi.Auth)
+
+	// un-authenticated client
+	if _, err := authapi.UserAdd(context.TODO(), "foo", "bar"); err != rpctypes.ErrUserNotFound {
+		t.Fatalf("expected %v, got %v", rpctypes.ErrUserNotFound, err)
+	}
+
+	// wrong id or password
+	if _, err := integration.NewClientV3Auth(clus.Members[0], "foo", "bar"); err != rpctypes.ErrAuthFailed {
```
Comment (heyitsanthony) [inline clientv3/integration/user_test.go]: can this not add any new functions to `integration`? it seems like if it's defining auth stuff the client should be created off a user-supplied config...

```
cli := clus.Client(0)
cfg := clientv3.Config{Endpoints: cli.Endpoints()}
cfg.Username, cfg.Password = "foo", "bar" 
if _, err := clientv3.New(cfg); err != rpctypes.ErrAuthFailed {
...
}
...
```

**Comment by gyuho** [comment]
@heyitsanthony Fixed. PTAL

File: clientv3/integration/user_test.go [hunk 2]
```diff
@@ -51,3 +52,55 @@ func TestUserError(t *testing.T) {
...
+
+	authapi := clus.RandClient()
+	authSetupRoot(t, authapi.Auth)
+
+	// un-authenticated client
+	if _, err := authapi.UserAdd(context.TODO(), "foo", "bar"); err != rpctypes.ErrUserNotFound {
+		t.Fatalf("expected %v, got %v", rpctypes.ErrUserNotFound, err)
+	}
+
+	// wrong id or passwordcli := clus.Client(0)
```
Comment (heyitsanthony) [inline clientv3/integration/user_test.go]: busted comment

**Comment by heyitsanthony** [comment]
lgtm after comment fix

**Comment by mitake** [comment]
@gyuho thanks!

**Pull request merged**